packages/expo-cli/src/appleApi/authenticate.ts

```diff
@@ -7,6 +11,99 @@ import prompt from '../prompt';
 import { nonEmptyInput } from '../validators';
 import { runAction, travelingFastlane } from './fastlane';
 
+const NO_STORE_PASSWORD = getenv.boolish('EXPO_NO_STORE_PASSWORD', false);
+
+function getKeychainServiceName(appleId: string): string {
+  return `deliver.${appleId}`;
+}
+
+async function deletePasswordAsync({
+  appleId,
+}: Pick<AppleCredentials, 'appleId'>): Promise<boolean> {
+  // Mac only right now.
+  if (process.platform !== 'darwin') {
+    return false;
+  }
+  const keychainService = getKeychainServiceName(appleId);
+  return new Promise((resolve, reject) => {
+    keychain.deletePassword(
+      { account: appleId, service: keychainService, type: 'internet' },
+      (error: Error) => {
+        if (error) {
+          if (error.message.match(/Could not find password/)) {
```

Comment:
nit: put `/Could not find password/` into a constant since we are using it multiple places